src/gpio/g4.rs

```diff
@@ -0,0 +1,369 @@
+    PC5: (pc5, 5, [2, 3, 6, 7, 15]),
+    PC6: (pc6, 6, [2, 4, 6, 15]),
+    PC7: (pc7, 7, [2, 4, 6, 15]),
+    PC8: (pc8, 8, [2, 4, 8, 15]),
+    PC9: (pc9, 9, [2, 4, 5, 6, 8, 15]),
+    PC10: (pc10, 10, [4, 5, 6, 7, 15]),
+    PC11: (pc11, 11, [4, 5, 6, 7, 8, 15]),
+    PC12: (pc12, 12, [4, 6, 7, 14, 15]),
+    PC13: (pc13, 13, [2, 4, 6, 15]),
+    PC14: (pc14, 14, [15]),
+    PC15: (pc15, 15, [15]),
+]);
+
+#[cfg(feature = "gpio-g43x")]
+gpio!(GPIOD, gpiod, PD, 'D', PDn, [
+    PD0: (pd0, 0, [6, 9, 15]),
+    PD1: (pd1, 1, [4, 6, 9, 15]),
+    PD2: (pd2, 2, [2, 4, 15]),
+    PD3: (pd3, 3, [2, 7, 15]),
+    PD4: (pd4, 4, [2, 7, 15]),
+    PD5: (pd5, 5, [7, 15]),
+    PD6: (pd6, 6, [2, 3, 7, 13, 15]),
+    PD7: (pd7, 7, [2, 7, 15]),
+    PD8: (pd8, 8, [7, 15]),
+    PD9: (pd9, 9, [7, 15]),
+    PD10: (pd10, 10, [7, 15]),
+    PD11: (pd11, 11, [7, 15]),
+    PD12: (pd12, 12, [2, 7, 15]),
+    PD13: (pd13, 13, [2, 15]),
+    PD14: (pd14, 14, [2, 15]),
+    PD15: (pd15, 15, [2, 6, 15]),
+]);
+
+#[cfg(feature = "gpio-g43x")]
+gpio!(GPIOE, gpioe, PE, 'E', PEn, [
+    PE0: (pe0, 0, [2, 4, 7, 15]),
+    PE1: (pe1, 1, [4, 7, 15]),
+    PE2: (pe2, 2, [0, 2, 3, 13, 15]),
+    PE3: (pe3, 3, [0, 2, 13, 15]),
+    PE4: (pe4, 4, [0, 2, 3, 13, 15]),
+    PE5: (pe5, 5, [0, 2, 3, 13, 15]),
+    PE6: (pe6, 6, [0, 3, 13, 15]),
+    PE7: (pe7, 7, [2, 13, 15]),
+    PE8: (pe8, 8, [2, 13, 15]),
+    PE9: (pe9, 9, [2, 13, 15]),
+    PE10: (pe10, 10, [2, 13, 15]),
+    PE11: (pe11, 11, [2, 15]),
+    PE12: (pe12, 12, [2, 15]),
+    PE13: (pe13, 13, [2, 15]),
+    PE14: (pe14, 14, [2, 6, 15]),
+    PE15: (pe15, 15, [2, 6, 7, 15]),
+]);
+
+#[cfg(feature = "gpio-g43x")]
+gpio!(GPIOF, gpiof, PF, 'F', PFn, [
+    PF0: (pf0, 0, [4, 5, 6, 15]),
+    PF1: (pf1, 1, [5, 15]),
+    PF2: (pf2, 2, [4, 15]),
+    PF9: (pf9, 9, [3, 5, 13, 15]),
+    PF10: (pf10, 10, [3, 5, 13, 15]),
+]);
+
+#[cfg(feature = "gpio-g43x")]
+gpio!(GPIOG, gpiog, PG, 'G', PGn, [
+    PG10: (pg10, 10, [0, 15]),
+]);
+
+#[cfg(feature = "gpio-g47x")]
+gpio!(GPIOA, gpioa, PA, 'A', PAn, [
+    PA0: (pa0, 0, [1, 2, 7, 8, 9, 10, 14, 15]),
+    PA1: (pa1, 1, [0, 1, 2, 7, 9, 15]),
+    PA2: (pa2, 2, [1, 2, 7, 8, 9, 10, 12, 14, 15]),
+    PA3: (pa3, 3, [1, 2, 3, 7, 9, 10, 12, 13, 15]),
+    PA4: (pa4, 4, [2, 5, 6, 7, 13, 15]),
+    PA5: (pa5, 5, [1, 2, 5, 14, 15]),
+    PA6: (pa6, 6, [1, 2, 4, 5, 6, 8, 10, 12, 15]),
+    PA7: (pa7, 7, [1, 2, 4, 5, 6, 8, 10, 14, 15]),
+    PA8: (pa8, 8, [0, 2, 4, 5, 6, 7, 8, 10, 11, 12, 13, 14, 15]),
+    PA9: (pa9, 9, [2, 4, 5, 6, 7, 8, 9, 10, 13, 14, 15]),
+    PA10: (pa10, 10, [1, 3, 4, 5, 6, 7, 8, 10, 11, 12, 13, 14, 15]),
+    PA11: (pa11, 11, [5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15]),
+    PA12: (pa12, 12, [1, 5, 6, 7, 8, 9, 10, 11, 13, 14, 15]),
+    PA13: (pa13, 13, [0, 1, 3, 4, 5, 7, 10, 13, 15], super::Debugger),
+    PA14: (pa14, 14, [0, 1, 3, 4, 5, 6, 7, 13, 15], super::Debugger),
+    PA15: (pa15, 15, [0, 1, 2, 4, 5, 6, 7, 8, 9, 11, 13, 14, 15], super::Debugger),
```

Comment:
Nice! This resolves https://github.com/stm32-rs/stm32g4xx-hal/issues/208#issuecomment-2918959470